enhancements/two-nodes-openshift/2no.md

```diff
@@ -113,32 +119,57 @@ In both cases, the control-plane's dependence on etcd will cause it to respond w
-2. What mechanism will pacemaker use to get access to the secret linked from the BareMetalHost CRD?
+###### Configuring Fencing Via 2NO Resource Agent Remediation Operator
+Fencing setup is the last important aspect of the cluster installation. For the cluster installation to be successful, fencing should be configured and active before we declare the installation successful. In order to do this, baseboard management console (BMC) credentials need to be made available to the control-plane nodes as part of pacemaker initialization.
+To ensure rapid fencing using pacemaker, we will collect RedFish details (address, username, and **password**) for each node via the install-config.
+This will take a format similar to that of the [Baremetal Operator](https://docs.openshift.com/container-platform/4.17/installing/installing_bare_metal_ipi/ipi-install-installation-workflow.html#bmc-addressing_ipi-install-installation-workflow).
+We will create a new CRD that stores the BMC address as well as a credentialsName to be managed by the new operator. This will resemble the BMC specification used by the [BareMetalHost](https://docs.openshift.com/container-platform/4.17/rest_api/provisioning_apis/baremetalhost-metal3-io-v1alpha1.html#spec-bmc) CRD.
 
-If available via the SaaS offering (not confirmed), ZTP may be evaluated as a future offering. This will need further evaluation to ensure passwords are appropriately handled.
+BMC information can be used to change the power state of a baremetal machine, so it's critically important that we ensure that pacemaker is the **only entity** responsible for these operations to prevent conflicting requests to change the machine state.
+This means that we need to ensure that there are protections in the Baremetal Operator (BMO) to prevent control plane nodes from having power management enabled in a two-node topology. Additionally, optional operators like Node Health Check, Self Node Remediation, and Fence Agents Remediation must have the same considerations but these should not be present during installation.
 
-Everything else about cluster creation will be an opaque implementation detail not exposed to the user.
+For a two-node cluster to be successful, we need to ensure the following:
+1. The new operator must be able to watch for changes to the new CRD and synchronize the BMC secrets for RHEL-HA once during initialization and again whenever the CRD is updated.
+3. The new operator should run the `pcs` configuration command on the nodes to enable the fencing agent and report an error to the user if the fencing configuration isn't successful. In the case of installation, this means that the operator becomes degraded to ensure that the installation fails.
+4. The new operator should periodically check that the fencing agent is healthy to ensure that the user can be notified in case the credential is rotated and needs to be updated.
```

Comment:
I think this has some major implications I'm not sure we've discussed.
The two things from the top of my mind are:
- The 2no operator will need to be running beyond the setup scope
- We assume fence agents needs external monitoring

Reply:
@mshitrit pacemaker does monitoring of the fencing devices on a regular base to report to the administrator that something is off.

However, it is incorrect to assume that the monitor operation will fail only because credentials needs to be updated.

A monitoring operation can fail for multiple reasons, including an admin performing a BMC firmware update that don´t need a server reboot, or a network blip between the host and the BMC, etc. etc.